### PR TITLE
Add Gurobi pool output.

### PR DESCRIPTION
**Pull request by WimVanRoy**
Allow the users to access all solutions from the solution pool in Gurobi.

### CODE REVIEW

File: casadi/interfaces/gurobi/gurobi_interface.cpp [hunk 1]
```diff
@@ -159,6 +159,8 @@ namespace casadi {
     casadi_assert(!flag && m->env, "Failed to create GUROBI environment. Flag: " + str(flag)
       + ":" + GRBgeterrormsg(m->env));
 
+    m->pool_sol_nr = 0;
```
Comment (jgillis) [inline casadi/interfaces/gurobi/gurobi_interface.cpp:162]: Is this supposed to be an option?

Comment (WimVanRoy) [reply casadi/interfaces/gurobi/gurobi_interface.cpp:162]: Nope, this is one of the outputs.
We could add an option to receive the pool or not?

Comment (jgillis) [reply casadi/interfaces/gurobi/gurobi_interface.cpp:162]: Indeed I misread. This needs to be optional. However, Gurobi may probably already have an option for this already?

Comment (WimVanRoy) [reply casadi/interfaces/gurobi/gurobi_interface.cpp:162]: Yes, you can limit the number of solutions in the solution pool, but the minimum is always 1.

**Pull request merged**